android/java/org/chromium/chrome/browser/externalnav/BraveExternalNavigationHandler.java

```diff
@@ -45,8 +45,12 @@ public OverrideUrlLoadingResult shouldOverrideUrlLoading(ExternalNavigationParam
     protected OverrideUrlLoadingResult startActivity(Intent intent, boolean requiresIntentChooser,
             QueryIntentActivitiesSupplier resolvingInfos, ResolveActivitySupplier resolveActivity,
             GURL browserFallbackUrl, GURL intentDataUrl, ExternalNavigationParams params) {
-        if (ContextUtils.getAppSharedPreferences().getBoolean(
-                    BravePrivacySettings.PREF_APP_LINKS, true)) {
+        boolean isYoutubeDomain = intentDataUrl.domainIs(BraveConstants.YOUTUBE_DOMAIN);
+        if ((isYoutubeDomain
+                        && !BravePrefServiceBridge.getInstance().getPlayYTVideoInBrowserEnabled())
```

Comment:
@tapanmodh i think we should try using prefs directly instead of BravePrefServiceBridge. i will create an issue to address it separately.

Reply:
@deeppandya i have created issue for replace `BravePrefServiceBridge` for Brave News https://github.com/brave/brave-browser/issues/33464. I think we can address this in that issue as well.